internal/reconciliations/istio_resources/envoy_filter_allow_partial_referer.go

```diff
@@ -58,6 +59,12 @@ func (e EnvoyFilterAllowPartialReferer) reconcile(ctx context.Context, k8sClient
 	}
 
 	spec := envoyFilter.Object["spec"]
+	versionedLabels := envoyFilter.GetLabels()
```

Comment:
The way how reconciliation is implemented right now it was easier to copy over the code rather than making code more reusable. I can't think of instant solution. I'd suggest leaving it as-is and if the time for changing reconciliation logic comes eventually, or labeling gets changed (most likely) then we can revisit this part of code again.